Primo.js

```diff
@@ -363,37 +358,184 @@ function importPNX(text) {
+		numPages.push(m[1].replace(/ /g, '')
+			.replace(/[\-,]/g,'+')
+			.toLowerCase() // for Roman numerals
+		);
+	}
+	return numPages.join('; ');
 }/** BEGIN TEST CASES **/
 var testCases = [
 	{
 		"type": "web",
-		"defer": true,
-		"url": "http://solo.bodleian.ox.ac.uk/primo_library/libweb/action/dlDisplay.do?docId=oxfaleph010311597&vid=OXVU1",
+		"url": "http://searchit.princeton.edu/primo_library/libweb/action/dlDisplay.do?docId=PRN_VOYAGER2778598&vid=PRINCETON&institution=PRN&showPNX=true",
 		"items": [
 			{
 				"itemType": "book",
+				"title": "China and foreign missionaries.",
 				"creators": [
 					{
-						"firstName": "Jürgen",
-						"lastName": "Pütz",
-						"creatorType": "author"
+						"lastName": "Great Britain. Foreign Office",
+						"creatorType": "author",
+						"fieldMode": 1
 					}
 				],
+				"date": "1860",
+				"language": "eng",
+				"libraryCatalog": "Primo",
+				"publisher": "London",
+				"attachments": [],
+				"tags": [
+					"China Foreign relations Great Britain.",
+					"China Religion.",
+					"Great Britain Foreign relations China.",
+					"Missions China."
+				],
 				"notes": [],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "http://purdue-primo-prod.hosted.exlibrisgroup.com/primo_library/libweb/action/dlDisplay.do?vid=PURDUE&docId=PURDUE_ALMA21505315560001081&fn=permalink",
+		"items": [
+			{
+				"itemType": "book",
+				"title": "War",
+				"creators": [
+					{
+						"firstName": "Lawrence",
+						"lastName": "Freedman",
+						"creatorType": "author"
+					}
+				],
+				"date": "1994",
+				"ISBN": "0192892541",
+				"abstractNote": "Experience of war -- Causes of war -- War and the military establishment -- Ethics of war -- Strategy -- Total war and the great powers -- Limited war and developing countries., \"War makes headlines and history books. It has shaped the international system, prompted social change, and inspired literature, art, and music. It engenders some of the most intense as well as the most brutal human experiences, and it raises fundamental questions of human ethics.\" \"The ubiquitous, contradictory, and many-sided character of war is fully reflected in this reader. It addresses a wide range of questions: What are the causes of war? Which strategic as well as moral principles guide its conduct, and how have these changed? Has total war become unthinkable? What is the nature of contemporary conflict? How is war experienced by those on the front line?\" \"These and other key issues are examined through a variety of writings. Drawing on sources from numerous countries and disciplines, this reader includes accounts by generals, soldiers, historians, strategists, and poets, who consider conflicts from the Napoleonic Wars to Vietnam and Bosnia. The writing not only of great strategic thinkers but also of ordinary soldiers illustrates both the theory and the experience of war in its many guises.\"--BOOK JACKET.",
+				"callNumber": "U21.2",
```

Comment:
A seperate PR/issue is a good idea. One remark/reminder (such that we don't forget it): We have here in primo the <a href="http://en.wikipedia.org/wiki/Library_of_Congress_Classification">LCC</a> (classification) which is not the same as the <a href="http://en.wikipedia.org/wiki/Library_of_Congress_Control_Number">LCCN</a> (id).